storm-core/src/clj/org/apache/storm/testing.clj

```diff
@@ -242,11 +297,12 @@
         (.stop (:nimbus-thrift-server cluster-map))
         (catch Exception e (log-message "failed to stop thrift")))
       ))
-  (.close (:state cluster-map))
-  (.disconnect (:storm-cluster-state cluster-map))
-  (doseq [s @(:supervisors cluster-map)]
-    (.shutdownAllWorkers s nil ReadClusterState/THREAD_DUMP_ON_ERROR)
-    (.close s))
+  (if (:state cluster-map) (.close (:state cluster-map)))
```

Comment:
In the current code there are multiple "constructors" for the cluster-map.  It was not clear to me that I added state everywhere so I decided to be defensive in my code (especially because I saw some tests that manipulate maps that I though should be internal).  In the next pull request that replaces testing.clj etc all of this gets cleaned up.